### PR TITLE
OCPBUGS-15583: fix nodeStatusUpdateFrequency

### PR DESCRIPTION
**Pull request by cdoern**
if the customer does not specify nodeStatusUpdateFrequency, we should still set it to the default value. If we do not, it seems that it gets set to zero.

While the 0 will be eventually translated to 10 so the effect is the same, our handling causes unintended diffs in kubeletConfigs in the intermediary which seems to confuse the MCO in choosing the proper machineConfig to update to.

### CODE REVIEW

File: test/e2e/kubeletcfg_test.go [hunk 1]
```diff
@@ -117,46 +152,76 @@ func runTestWithKubeletCfg(t *testing.T, testName, regexKey, expectedConfVal1, e
...
-	cleanupKcFunc2 := createKcWithConfig(t, cs, kcName2, poolName, &kc2.Spec, "1")
-	// wait for the second kubelet config to show up
-	kcMCName2, err := getMCFromKubeletCfg(t, cs, kcName2)
-	require.Nil(t, err, "failed to render machine config from second container runtime config")
-	// ensure the second kubelet config update rolls out to the pool
-	helpers.WaitForConfigAndPoolComplete(t, cs, poolName, kcMCName2)
-	// verify value was changed to match that of the first kubelet config
-	secondConfValue := getValueFromKubeletConfig(t, cs, node, regexKey, kubeletPath)
-	require.Equal(t, secondConfValue, expectedConfVal2, "value in kubelet config not updated as expected")
+	if *kc1.Spec.AutoSizingReserved {
```
Comment (harche) [inline test/e2e/kubeletcfg_test.go:165]: Thanks for adding the test for auto node sizing.

File: pkg/controller/kubelet-config/helpers.go [hunk 2]
```diff
@@ -155,7 +155,7 @@ func updateOriginalKubeConfigwithNodeConfig(node *osev1.Node, originalKubeletCon
 	case osev1.DefaultUpdateDefaultReaction:
```
Comment (sairameshv) [inline pkg/controller/kubelet-config/helpers.go:155]:
```suggestion
	case osev1.DefaultUpdateDefaultReaction, emptyInput:
```

File: pkg/controller/kubelet-config/helpers.go [hunk 3]
```diff
@@ -155,7 +155,7 @@ func updateOriginalKubeConfigwithNodeConfig(node *osev1.Node, originalKubeletCon
...
 		originalKubeletConfig.NodeStatusUpdateFrequency = metav1.Duration{Duration: osev1.DefaultNodeStatusUpdateFrequency}
 	case emptyInput:
-		return nil
+		originalKubeletConfig.NodeStatusUpdateFrequency = metav1.Duration{Duration: osev1.DefaultNodeStatusUpdateFrequency}
```
Comment (sairameshv) [inline pkg/controller/kubelet-config/helpers.go:158]: nit: This case can be removed by appending the `emptyInput` to the above case as suggested

File: templates/master/01-master-kubelet/_base/files/kubelet.yaml [hunk 4]
```diff
@@ -25,6 +25,7 @@ contents:
     serializeImagePulls: false
     staticPodPath: /etc/kubernetes/manifests
     systemCgroups: /system.slice
+    nodeStatusUpdateFrequency: 10s
```
Comment (sairameshv) [inline templates/master/01-master-kubelet/_base/files/kubelet.yaml:28]: @cdoern , When this base kubelet template is already getting updated, I don't think we really need to make a change to this [case](https://github.com/openshift/machine-config-operator/pull/3784/files#diff-4a2d0a4442e10d889a927322ad7311c67b22b259622067b52cbda7d2ece3ebefR158) . WDYT?

File: pkg/controller/kubelet-config/helpers.go (hunk 3 above)
Comment (cdoern) [reply pkg/controller/kubelet-config/helpers.go:158]: I think it might be best to leave the case for empty just in case a user somehow messes with their config?

File: templates/master/01-master-kubelet/_base/files/kubelet.yaml (hunk 4 above)
Comment (cdoern) [reply templates/master/01-master-kubelet/_base/files/kubelet.yaml:28]: I think it might be best to leave the case for empty just in case a user somehow messes with their config?

Comment (sairameshv) [reply templates/master/01-master-kubelet/_base/files/kubelet.yaml:28]: /hold
In my opinion, if you are already making a change to the base templates, you need not make any change to the `updateOriginalKubeConfigwithNodeConfig` function.
One of the reasons being:
Let's say in future, based on a new requirement (may be), we decide to set the `DefaultNodeStatusUpdateFrequency` to be `20s`, the `emptyInput` should set nothing and the underlying Kubelet config sets the `nodeStatusUpdateFrequency` to `10s`(as set in base template) and not `20s`
Basically, default `nodeStatusUpdateFrequency` need not depend on the `workerLatencyProfile` field's default value.

Any thoughts @harche @yuqi-zhang ?

**Review by sairameshv — COMMENTED** [review]
Please feel free to unhold if you think all the review comments are addressed and resolved.

**Comment by cdoern** [comment]
@sairameshv I do not feel too strongly about keeping the empty changes, so I made your suggestions. As long as upgrade tests pass, I am ok with it!
